client/packages/system/src/Manage/Reports/hooks/useAllReportVersionsList.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  useQuery,
+  SortBy,
+  ReportSortFieldInput,
+  ReportFilterInput,
+  useIntlUtils,
+  useTranslation,
+  useNotification,
+  LIST_KEY,
+} from '@openmsupply-client/common';
+import {  ALLREPORTVERSIONS } from './keys';
+import { ReportRowFragment } from 'packages/system/src/Report/index.js';
+import { useAllReportVersionsGraphQL } from '../api/useReportsGraphQL';
+
+export type ReportListParams = {
+  filterBy: ReportFilterInput | null;
+  sortBy?: SortBy<ReportRowFragment>;
+  first?: number;
+  offset?: number;
+};
+
+export const useAllReportVersionsList = ({
+  queryParams,
+}: {
+  queryParams?: ReportListParams;
+}) => {
+
+    // QUERY
+
+    const { data, isLoading, isError } = useGetList(queryParams);
+
+    return {
+        query: {data, isLoading, isError},
+    };
```

Comment:
I hope I'm using the correct new pattern here for these hooks.

Later if needed we could add mutations (in particular to be able to set reports to be active)

Reply:
On track! 🔥 🔥 🔥